### PR TITLE
Add remove header function in HttpRequest

### PR DESCRIPTION
**Pull request by Christoffer-Cortes**
### Applicable Issues
Needed in the integration test for eiffel-intelligence-frontend.

### Sign-off
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Christoffer Cortes Sjöwall, [christoffer.cortes.sjowall@ericsson.com](mailto:christoffer.cortes.sjowall@ericsson.com)

### CODE REVIEW

File: src/main/java/com/ericsson/ec/JenkinsManager.java [hunk 1]
```diff
@@ -105,25 +104,23 @@ public boolean createJob(String jobName, String jobXmlData) throws Exception {
             throw new Exception("A job is no one! Jenkins do not like no one!");
         }
 
-        httpRequest.setBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding)
-                .addHeader("Content-type", MediaType.APPLICATION_XML).addHeader("Jenkins-Crumb", crumb)
-                .addParam("name", jobName).setBody(jobXmlData).setEndpoint("/createItem");
+        httpRequest.setBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding).addHeader("Content-type", MediaType.APPLICATION_XML)
```
Comment (e-edling-ericsson) [inline src/main/java/com/ericsson/ec/JenkinsManager.java]: Chains these calls under eachother, easier to see whats happening and also shorter lines :)

File: src/main/java/com/ericsson/ec/JenkinsManager.java [hunk 2]
```diff
@@ -224,15 +217,15 @@ public JSONObject getJenkinsBuildStatusData(String jobName, Integer buildNumber)
         }
         String endpoint = "/job/" + jobName + "/" + buildNumberString + "/api/json";
 
-        httpRequest.setBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding)
-                .addHeader("Content-type", MediaType.APPLICATION_JSON).setEndpoint(endpoint);
+        httpRequest.setBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding).addHeader("Content-type", MediaType.APPLICATION_JSON)
```
Comment (e-edling-ericsson) [inline src/main/java/com/ericsson/ec/JenkinsManager.java]: Chain these under eachother

File: src/main/java/com/ericsson/ec/JenkinsManager.java [hunk 3]
```diff
@@ -272,16 +263,14 @@ public boolean deleteJob(String jobName) throws Exception {
 
         String endpoint = "/job/" + jobName + "/doDelete";
 
-        httpRequest.setBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding)
-                .addHeader("Content-type", MediaType.APPLICATION_JSON).addHeader("Jenkins-Crumb", crumb)
-                .setEndpoint(endpoint);
+        httpRequest.setBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding).addHeader("Content-type", MediaType.APPLICATION_JSON)
```
Comment (e-edling-ericsson) [inline src/main/java/com/ericsson/ec/JenkinsManager.java]: Same here

File: src/main/java/com/ericsson/ec/JenkinsManager.java [hunk 4]
```diff
@@ -300,15 +289,14 @@ public boolean pluginExists(String plugin) throws Exception {
         boolean pluginExists = false;
 
         HttpRequest httpRequest = new HttpRequest(HttpMethod.GET);
-        httpRequest.setBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding).addParam("depth", "1")
-                .addParam("wrapper", "plugins").setEndpoint("/pluginManager/api/json");
+        httpRequest.setBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding).addParam("depth", "1").addParam("wrapper", "plugins")
```
Comment (e-edling-ericsson) [inline src/main/java/com/ericsson/ec/JenkinsManager.java]: Same here

File: src/main/java/com/ericsson/ec/JenkinsManager.java [hunk 5]
```diff
@@ -349,17 +337,15 @@ public boolean installPlugin(String plugin, String version) throws Exception {
 
         String scriptData = "<jenkins><install plugin='" + plugin + "@" + version + "' /></jenkins>";
 
-        httpRequest.setBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding)
-                .addHeader("Content-type", MediaType.TEXT_XML).addHeader("Jenkins-Crumb", crumb).setBody(scriptData)
-                .setEndpoint("/pluginManager/installNecessaryPlugins");
+        httpRequest.setBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding).addHeader("Content-type", MediaType.TEXT_XML)
```
Comment (e-edling-ericsson) [inline src/main/java/com/ericsson/ec/JenkinsManager.java]: And here :)

File: src/main/java/com/ericsson/ec/JenkinsManager.java [hunk 6]
```diff
@@ -378,16 +364,14 @@ public boolean restartJenkins() throws Exception {
 
         ResponseEntity response = null;
         HttpRequest httpRequest = new HttpRequest(HttpMethod.POST);
-        httpRequest.setBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding)
-                .addHeader("Content-type", MediaType.APPLICATION_JSON).addHeader("Jenkins-Crumb", crumb)
-                .setEndpoint("/safeRestart");
+        httpRequest.setBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding).addHeader("Content-type", MediaType.APPLICATION_JSON)
```
Comment (e-edling-ericsson) [inline src/main/java/com/ericsson/ec/JenkinsManager.java]: etc etc

File: src/main/java/com/ericsson/ec/JenkinsManager.java (hunk 1 above)
Comment (Christoffer-Cortes) [reply src/main/java/com/ericsson/ec/JenkinsManager.java]: Done.

File: src/main/java/com/ericsson/ec/JenkinsManager.java [hunk 7]
```diff
@@ -87,8 +87,7 @@ public JenkinsManager(String protocol, String host, int port, String username, S
     }
```
Comment (a-breid-ericsson) [inline src/main/java/com/ericsson/ec/JenkinsManager.java]: Formatting in this file does not follow Eiffel formatting standards :)

Comment (Christoffer-Cortes) [reply src/main/java/com/ericsson/ec/JenkinsManager.java]: I updated the eiffel code style formatter in confluence to support chained method calls.

Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/ec/JenkinsManager.java]: The long javadoc comments.

File: src/main/java/com/ericsson/ec/helpers/JenkinsXmlData.java [hunk 8]
```diff
@@ -187,20 +180,17 @@ private void validatePropertiesObject(String parametertypeKey) {
             properties.put(HUDSON_PARAMETERS_DEFINITION_KEY, parametersDefinitionProperty);
         }
 
-        boolean hasParameterDefinitions = properties.getJSONObject(HUDSON_PARAMETERS_DEFINITION_KEY)
-                .has(PARAMETER_DEFINITION_KEY);
+        boolean hasParameterDefinitions = properties.getJSONObject(HUDSON_PARAMETERS_DEFINITION_KEY).has(PARAMETER_DEFINITION_KEY);
```
Comment (a-breid-ericsson) [inline src/main/java/com/ericsson/ec/helpers/JenkinsXmlData.java]: This line maybe should not be so long?

File: src/main/java/com/ericsson/ec/JenkinsManager.java [hunk 9]
```diff
@@ -349,17 +351,19 @@ public boolean installPlugin(String plugin, String version) throws Exception {
...
         success = response.getStatusCode() == HttpStatus.SC_MOVED_TEMPORARILY;
 
         if (!success) {
-            String message = "Failed to add a plugin with name " + plugin + " and version " + version
-                    + " to Jenkins. Response code: " + response.getStatusCodeValue() + " and body: "
-                    + response.getBody();
+            String message = "Failed to add a plugin with name " + plugin + " and version " + version + " to Jenkins. Response code: "
+                    + response.getStatusCodeValue() + " and body: " + response.getBody();
             throw new Exception(message);
         }
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ec/JenkinsManager.java]: Do we check if Plugin is successfully installed and usable when this function has returned?
It is no meaning to execute tests if required plugin is not installed and/or is not working.

File: src/main/java/com/ericsson/ec/JenkinsManager.java [hunk 10]
```diff
@@ -466,23 +473,24 @@ private boolean verifyJenkinsRestart() throws Exception {
...
         do {
             try {
                 Thread.sleep(3000);
                 response = httpRequest.performRequest();
-                serverDownRecieved = response.getStatusCode() == HttpStatus.SC_SERVICE_UNAVAILABLE
-                        || serverDownRecieved;
+                serverDownRecieved = response.getStatusCode() == HttpStatus.SC_SERVICE_UNAVAILABLE || serverDownRecieved;
                 success = response.getStatusCode() == HttpStatus.SC_OK || success;
             } catch (Exception e) {
             }
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ec/JenkinsManager.java]: same comment here.

File: src/main/java/com/ericsson/ec/JenkinsManager.java [hunk 11]
```diff
@@ -466,23 +473,24 @@ private boolean verifyJenkinsRestart() throws Exception {
...
         } while (!success && stopTime > System.currentTimeMillis());
 
         if (!success) {
-            String message = "Could not verify tjat Jenkins started up correctly. Response code: "
-                    + response.getStatusCodeValue() + " and body: " + response.getBody();
+            String message = "Could not verify tjat Jenkins started up correctly. Response code: " + response.getStatusCodeValue() + " and body: "
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ec/JenkinsManager.java]: Spelling,,, "tjat" -> that.

File: src/main/java/com/ericsson/ec/JenkinsManager.java (hunk 7 above)
Comment (Christoffer-Cortes) [reply src/main/java/com/ericsson/ec/JenkinsManager.java]: We are using different versions of the format file then. The one I'm using is the one that from the link I sent to everyone in the team.

Comment (Christoffer-Cortes) [reply src/main/java/com/ericsson/ec/JenkinsManager.java]: I did notice that there are more files with chained methods though...

File: src/main/java/com/ericsson/ec/JenkinsManager.java [hunk 12]
```diff
@@ -137,7 +139,8 @@ public boolean createJob(String jobName, String jobXmlData) throws Exception {
     public boolean forceCreateJob(String jobName, String jobXmlData) throws Exception {
         try {
             deleteJob(jobName);
-        } catch (Exception e) {}
+        } catch (Exception e) {
+        }
```
[the remark this answers is not in the document]
Comment (Christoffer-Cortes) [reply src/main/java/com/ericsson/ec/JenkinsManager.java]: Removed two empty try catches.

File: src/main/java/com/ericsson/ec/JenkinsManager.java (hunk 7 above)
Comment (Christoffer-Cortes) [reply src/main/java/com/ericsson/ec/JenkinsManager.java]: The java convention file in confluence has a 150 character limit on comments.

File: src/main/java/com/ericsson/ec/helpers/JenkinsXmlData.java (hunk 8 above)
Comment (Christoffer-Cortes) [reply src/main/java/com/ericsson/ec/helpers/JenkinsXmlData.java]: Not according to the eiffel formatter in confluence. We are likely using different versions.

File: src/main/java/com/ericsson/ec/JenkinsManager.java (hunk 9 above)
Comment (Christoffer-Cortes) [reply src/main/java/com/ericsson/ec/JenkinsManager.java]: I did not write this. It checks a http status code after performing a request which is used to check success.

Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/ec/JenkinsManager.java]: there is a function to lookup installed plugins, the test should verify installation by using that function.

**Pull request merged**